sycl/source/detail/queue_impl.hpp

```diff
@@ -447,10 +447,8 @@ class queue_impl {
       bool IsKernel = Type == CG::Kernel;
       bool KernelUsesAssert = false;
       if (IsKernel)
-        KernelUsesAssert =
-            Handler.MKernel ? true
-                            : ProgramManager::getInstance().kernelUsesAssert(
-                                  Handler.MOSModuleHandle, Handler.MKernelName);
```

Comment:
Preferably this and the changes in sycl/include/CL/sycl/queue.hpp to be a separate PR as they seem fairly unrelated to the primary issues addressed by this PR.

Reply:
Yes, this is to get module splitting per source to work, which is partly why I am making this PR. I agree that considering what I have written in the description this is not in scope. I'll remove sections related to this and make another PR.